ddtrace/internal/_encoding.pyx

```diff
@@ -629,26 +627,39 @@ cdef class MsgpackEncoderV05(MsgpackEncoderBase):
         if ret != 0: return ret
         ret = self._pack_string(span.resource)
         if ret != 0: return ret
-        ret = msgpack_pack_uint64(&self.pk, span.trace_id or 0)
+
+        _ = span.trace_id
+        ret = msgpack_pack_uint64(&self.pk, _ if _ is not None else 0)
```

Comment:
Is the change from `span.trace_id or 0` to this conditional an optimization?

Reply:
Yes. According to Cython annotate this should be "lighter"

Reply:
a comment would be nice on what optimization we are getting from this pattern